pipeline/test/services/service-cluster/testOpensearch.sh

```diff
@@ -0,0 +1,426 @@
+
+    for roleMapping in "${configuredMappings[@]}"; do
+        configured_mapping_name=$(echo "$roleMapping" | yq4 e '.mapping_name' -)
+        configured_users=$(echo "$roleMapping" | yq4 e '.definition.users[]' -)
+        res=$(echo "$rolesmapping" | jq ".$configured_mapping_name")
+        if [[ $res != "null" ]]; then
+            users=$(echo "$rolesmapping" | jq -r ".$configured_mapping_name.users[]")
+            if [[ "$users" != "$configured_users" ]]; then
+                no_error=false
+                debug_msg+="[ERROR] Missing users < ${configured_users//$'\n'/ }> != < $users >, for role mapping: $configured_mapping_name \n"
+            fi
+        else
+            no_error=false
+            debug_msg+="[ERROR] Missing role mapping_name ${roleMapping} \n"
+        fi
+    done
+
+    if $no_error; then
+        echo "success ✔"
+        echo "[DEBUG] User roles are configured correctly"
+    else
+        echo "failure ❌"
+        echo -ne "$debug_msg"
+    fi
+}
+
+check_opensearch_ism() {
+    echo -ne "Checking opensearch Index State Management ... "
+    no_error=true
+    debug_msg=""
+
+    default_policies_enabled=$(yq4 -e '.opensearch.ism.defaultPolicies' "${config['config_file_sc']}")
+    if [[ $default_policies_enabled == "true" ]]; then
+        default_policies=("kubernetes" "kubeaudit" "authlog" "other")
+        for policy in "${default_policies[@]}"; do
+            res=$(curl -w "%{http_code}" -o /dev/null -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_plugins/_ism/policies/${policy}")
+            if [[ $res != "200" ]]; then
+                no_error=false
+                debug_msg+="[ERROR] Missing default policy : ${policy}\n"
+            fi
+        done
+    fi
+
+    # additional_policies=$(yq4 -e '.opensearch.ism.additionalPolicies | keys' "${config['config_file_sc']}")
+    readarray -t additional_policies < <(yq4 e -o=j -I=0 '.opensearch.ism.additionalPolicies | keys' "${config['config_file_sc']}")
+
+    additional_policies=("${additional_policies//,/ }")
+    additional_policies=("${additional_policies##[}")
+    additional_policies=("${additional_policies%]}")
+    # additional_policies=($additional_policies)
+    read -ra additional_policies <<<"${additional_policies[@]}"
+
+    for policy in "${additional_policies[@]}"; do
+        policy_name=$(echo "$policy" | awk -F'.' '{print $1}' | tr '"' ' ')
+        res=$(curl -w "%{http_code}" -o /dev/null -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_plugins/_ism/policies/${policy_name}")
+        if [[ $res != "200" ]]; then
+            no_error=false
+            debug_msg+="[ERROR] Missing additional policy : ${policy_name}\n"
+        fi
+    done
+
+    indices_data=$(curl -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_cat/indices" | awk '{print $3}' | tr '\n' ' ')
```

Comment:
Try to prefilter and just get what you need instead of fetching all and inspecting each as it becomes really slow with many indices. (25 seconds for an empty production-like environment.)
```suggestion
    indices_data=$(curl -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_cat/aliases?format=json" | jq -r '.[] | select(.is_write_index == "true") | .index')
```
And yes `is_write_index` through CAT is apparently a string.

Reply:
Thanks a lot, it's changed now